service/service.go

```diff
@@ -138,3 +150,43 @@ func (s *Service) Boot() {
 		go s.KVMClusterConfigFramework.Boot()
 	})
 }
+
+func newBaseClusterConfig(f *flag.Flag, v *viper.Viper) (*cluster.Config, error) {
+	networkIP, apiServerIP, err := parseClusterIPRange(v.GetString(f.Guest.Cluster.Kubernetes.API.ClusterIPRange))
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	clusterConfig := &cluster.Config{
+		CertTTL: v.GetString(f.Guest.Cluster.Vault.Certificate.TTL),
+		IP: cluster.IP{
+			API:   apiServerIP,
+			Range: networkIP,
+		},
+	}
+
+	return clusterConfig, nil
+}
+
+func parseClusterIPRange(ipRange string) (net.IP, net.IP, error) {
+	_, cidr, err := net.ParseCIDR(ipRange)
+	if cidr == nil || err != nil {
+		return nil, nil, microerror.Maskf(invalidConfigError, "invalid Kubernetes ClusterIPRange")
+	}
+
+	ones, bits := cidr.Mask.Size()
+	if bits != 32 {
+		return nil, nil, microerror.Maskf(invalidConfigError, "Kubernetes ClusterIPRange CIDR must be an IPv4 range")
+	}
+
+	// Node gets /24 from Kubernetes and each POD receives one IP from this
+	// block. Therefore CIDR block must be at least /24.
+	if ones > 24 {
+		return nil, nil, microerror.Maskf(invalidConfigError, "Kubernetes ClusterIPRange CIDR network block must be at least /24")
```

Comment:
at most :)

Reply:
I know this is silly, but I bit the bullet :)

I'm thinking about the network size. The less there are bits, the larger the network is in terms of possible addresses.

In [CIDR Wikipedia page](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) there's also similar case:

> In routed subnets larger than /31 or /32, the number of available host addresses is usually reduced by two, namely the largest address, which is reserved as the broadcast address, and the smallest address, which identifies the network itself.

WDYT?

Reply:
Yeah, makes sense. Maybe even intuitive for some ppl :D